derive_builder/tests/bounds_generation.rs

```diff
@@ -0,0 +1,55 @@
+#[macro_use]
+extern crate derive_builder;
+
+/// A struct that deliberately doesn't implement `Clone`.
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct Dolor(String);
+
+/// Notice that this type derives Builder without disallowing
+/// `Lorem<Dolor>`.
+#[derive(Debug, Clone, Builder, PartialEq, Eq)]
+#[builder(field(private), setter(into))]
+pub struct Lorem<T> {
+    ipsum: T,
+}
+
+fn make_default_lorem<T: Default>() -> Lorem<T> {
+    Lorem {
+        ipsum: Default::default()
+    }
+}
+
+fn make_u16_lorem(v: u16) -> Lorem<u16> {
+    Lorem {
+        ipsum: v
+    }
+}
+
+fn make_dolor_lorem(v: Dolor) -> Lorem<Dolor> {
+    Lorem {
+        ipsum: v
+    }
+}
+
+fn build_u16_lorem(v: u16) -> Lorem<u16> {
+    LoremBuilder::default()
+        .ipsum(v)
+        .build()
+        .unwrap()
+}
+
+#[test]
+fn u16_lorems() {
+    assert_eq!(make_u16_lorem(10), build_u16_lorem(10));
+}
+
+#[test]
+fn dolor_lorems() {
+    assert_eq!(make_default_lorem(), make_dolor_lorem(Dolor::default()));
```

Comment:
Hm, in a test summary `dolor_lorems` will not be very informative. :-)

Please try to describe the test property in the function name.

If I understand this correctly, it's actually a _compile-test_ in the sense that _hey this compiles now, but it didn't before_. This test is actually only asserting something (almost) arbitrary in order to show up in the test summary, right?

In that case I would prefer to turn it into something like this

```rust
#[test]
fn generic_field_without_clone_compiles() {
    // This is actually a compile test, if we can _name_ `LoremBuilder<Dolor>`, we are already fine.
    // But we will not be able to call any setters or build method on it, because that would
    // require the field to implement `Clone`.
    let _: LoremBuilder<Dolor>;
}
```